### PR TITLE
Cert Rotate Command for PG

### PR DESCRIPTION
**Pull request by SanjuPal01**
Signed-off-by: “SanjuPal01” <sanju.sanju@progress.com>
"Sahiba3108" <sgoyal@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Added the cert rotate command for HA service. Now user has to pass the certificate paths to cert-rotate cmd and it will rotate the certificates of all postgres nodes

### :chains: Related Resources
https://chefio.atlassian.net/browse/KINETICS-235
### :athletic_shoe: How to Build and Test the Change
Add this package in your manifest while deploying HA:
sahiba3108/automate-ha-deployment/0.1.0/20221028084038

Install this cli in your bastion host:
ssanju/automate-cli/0.1.0/20221031070243

Run the below command in bastion:
`chef-automate cert-rotate --public-cert client.pem --private-cert client-key.pem --root-ca root-ca.pem --pg`
(You can also use --postgres flag instead of pg)
This will rotate the public, private and root cert of all pg nodes.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2022-10-31 at 1 00 23 PM

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/certRotate.go [hunk 1]
```diff
@@ -129,6 +167,70 @@ func certRotate(cmd *cobra.Command, args []string) error {
...
+			file := []string{fileName}
+			tomlFilePath, err := getMergerTOMLPath(file, infra, timestamp, remoteService)
+			if err != nil {
+				return err
+			}
+
+			scriptCommands := fmt.Sprintf(BACKEND_COMMAND, dateFormat, remoteService, dateFormat, remoteService+timestamp)
+			if len(infra.Outputs.PostgresqlPrivateIps.Value) > 0 {
+				remoteIp := infra.Outputs.PostgresqlPrivateIps.Value[0]
+				err := copyFileToRemote(sskKeyFile, tomlFilePath, sshUser, remoteIp, remoteService+timestamp)
```
Comment (ArvinthC3000) [inline components/automate-cli/cmd/chef-automate/certRotate.go:188]: rebase to main and add 6th argument

File: components/automate-cli/cmd/chef-automate/certRotate.go [hunk 2]
```diff
@@ -129,6 +167,70 @@ func certRotate(cmd *cobra.Command, args []string) error {
...
+
+			frontendIps := append(infra.Outputs.AutomatePrivateIps.Value, infra.Outputs.ChefServerPrivateIps.Value...)
+			if len(frontendIps) == 0 {
+				return errors.New("No frontend IPs are found")
+			}
+			remoteService = "frontend"
+
+			scriptCommands = fmt.Sprintf(FRONTEND_COMMANDS, remoteService+timestamp, dateFormat)
+			for i := 0; i < len(frontendIps); i++ {
+				err := copyFileToRemote(sskKeyFile, filename_fe, sshUser, frontendIps[i], remoteService+timestamp)
```
Comment (ArvinthC3000) [inline components/automate-cli/cmd/chef-automate/certRotate.go:222]: Add 6th argument (default: false)

File: components/automate-cli/cmd/chef-automate/certRotate.go (hunk 1 above)
Comment (punitmundra) [inline components/automate-cli/cmd/chef-automate/certRotate.go:188]: add the args here

Comment (SanjuPal01) [reply components/automate-cli/cmd/chef-automate/certRotate.go]: Done

File: components/automate-cli/cmd/chef-automate/certRotate.go (hunk 2 above)
Comment (SanjuPal01) [reply components/automate-cli/cmd/chef-automate/certRotate.go]: Done

**Review by Sahiba3108 — APPROVED** [review]

File: components/automate-cli/cmd/chef-automate/certRotate.go (hunk 1 above)
Comment (SanjuPal01) [reply components/automate-cli/cmd/chef-automate/certRotate.go]: Done

**Comment by Sahiba3108** [comment]
https://github.com/chef/automate/pull/7522

**Pull request closed without merging**